### PR TITLE
Feature/pg tde

### PR DESCRIPTION
**Pull request by ygworldr**
#### Summary

PostgreSQL 12.3 Transparent Data Encryption changes.

#### Ticket Link

Fixes [WA-24](https://yt.worldr.com/issue/WA-24).

### CODE REVIEW

File: store/sqlstore/supplier.go [hunk 1]
```diff
@@ -216,6 +221,31 @@ func NewSqlSupplier(settings model.SqlSettings, metrics einterfaces.MetricsInter
...
+	conn.Write([]byte(topSecretKey))
+
+	var buf [1024]byte
+	_, err = conn.Read(buf[0:])
+	if err != nil {
+		mlog.Warn("Fatal error ", mlog.Err(err))
+		return 1
+	}
+
+	mlog.Info("Key listener quoth '" + string(buf[0:]) + "'")
```
Comment (Aurora12) [inline store/sqlstore/supplier.go:245]: Shouldn't it also return 1 or 0 (or error message/nothing)? So we know that the listener didn't have any issues...

File: api4/team_test.go [hunk 2]
```diff
@@ -2457,7 +2459,7 @@ func TestInviteGuestsToTeam(t *testing.T) {
 			"TeamDisplayName": th.BasicTeam.DisplayName,
 			"SiteName":        th.App.ClientConfig()["SiteName"]})
 
-	//Check if the email was send to the rigth email address
+	//Check if the email was send to the right email address
```
Comment (Aurora12) [inline api4/team_test.go:2462]: Another typo is in "was send" :)

File: store/sqlstore/supplier.go [hunk 3]
```diff
@@ -232,6 +263,14 @@ func setupConnection(con_type string, dataSource string, settings *model.SqlSett
 		if err == nil {
 			break
 		} else {
+
+			// Call key talker in case we need it.
+			mlog.Info("Sending key to the database on " + dbHostName + ":" + KEY_LISTENER_PORT)
+			if keyTalker(dbHostName+":"+KEY_LISTENER_PORT) != 0 {
+				mlog.Warn("Key talked failed.")
```
[the remark this answers is not in the document]
Comment (ygworldr) [reply store/sqlstore/supplier.go:270]: Fixed.

File: api4/team_test.go (hunk 2 above)
Comment (ygworldr) [reply api4/team_test.go:2462]: Fixed.

File: store/sqlstore/supplier.go [hunk 4]
```diff
@@ -216,6 +221,31 @@ func NewSqlSupplier(settings model.SqlSettings, metrics einterfaces.MetricsInter
...
+		return 1
+	}
+
+	topSecretKey := "882fb7c12e80280fd664c69d2d636913" // FIXME!!!!
+	conn.Write([]byte(topSecretKey))
+
+	var buf [1024]byte
+	_, err = conn.Read(buf[0:])
+	if err != nil {
+		mlog.Warn("Fatal error ", mlog.Err(err))
```
[the remark this answers is not in the document]
Comment (ygworldr) [reply store/sqlstore/supplier.go:241]: True. Copy and paste from old code. Fixed.

File: store/sqlstore/supplier.go (hunk 1 above)
Comment (ygworldr) [reply store/sqlstore/supplier.go:245]: Possibly. Although if you could send to it, the listener passed it onwards. It knows nothing of what is happening beyond that. I guess we could check if the answer is what we expect, but it either is going to be or we could not get there. There is a potential of the connection dropping just as we are passing the data but it is really unlikely to ever happen and we will detect that when the database fails to start.

**Comment by Aurora12** [comment]
obsolete

**Pull request closed without merging**